docs/api/navigation-history.md

```diff
@@ -17,13 +17,31 @@ Returns `Integer` - The index of the current page, from which we would go back/f
 
 * `index` Integer
 
-Returns `Object`:
-
-* `url` string - The URL of the navigation entry at the given index.
-* `title` string - The page title of the navigation entry at the given index.
+Returns [`NavigationEntry`](structures/navigation-entry.md) - Navigation entry at the given index.
 
 If index is out of bounds (greater than history length or less than 0), null will be returned.
 
 #### `navigationHistory.length()`
 
 Returns `Integer` - History length.
+
+#### `navigationHistory.deleteEntryAtIndex(index)`
+
+* `index` Integer
+
+Deletes the navigation entry at the given index. Can't delete entry at the "current active index".
+
+Returns `boolean` - Whether the navigation entry was removed from the webContents history.
+
+#### `navigationHistory.getHistory()`
+
+Returns [`NavigationEntry[]`](structures/navigation-entry.md) - WebContents complete history.
```

Comment:
It might be interesting to implement the [Iterator protocol](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols#the_iterator_protocol) for `navigationHistory` so that the following snippet would work
```ts
for (const entry of webContents.navigationHistory) {
  // ...
}
```
Not something to block this PR on tho.

Reply:
Here is the implementation of the restore method: https://source.chromium.org/chromium/chromium/src/+/main:content/browser/renderer_host/navigation_controller_impl.cc;l=823
You can find all references by clicking on the name of the function.